app/components/ChartDataFormatter/index.js

```diff
@@ -1,175 +1,155 @@
+    this.setState(
+      this._handleProps(this.props.options.tickFormatSettings || {})
+    );
   }
 
-  /**
-   * Send new tickFormat function to xAxis or yAxis, tooltip should inherit from yAxis.tickFormat
-   *
-   * @param string axis 'x' or 'y'
-   * @param function formatter
-   */
-  _update(axis, tickFormat, formatData) {
-    const axisName = `${axis}Axis`;
-    const newOpts = {
-      // merge new tickFormat() into options, e.g. options.yAxis.tickFormat
-      [axisName]: update(this.props.options[axisName] || {}, {
-        $merge: { tickFormat },
-      }),
-      // merge format builder data, e.g. options.tickFormatBuilder.yAxis
-      tickFormatBuilder: update(this.props.options.tickFormatBuilder || {}, {
-        $apply: (builder) => {
-          builder[axisName] = formatData; // eslint-disable-line no-param-reassign
-          return builder;
-        },
-      }),
-    };
-
-    // Apply formatting to single-series charts like pie chart, etc
-    if (!dataIsMultiSeries(this.props.data)) {
-      newOpts.valueFormat = tickFormat;
-    }
-
-    this.props.dispatch(actionTrigger(RECEIVE_CHART_OPTIONS, newOpts));
+  componentWillReceiveProps(nextProps) {
+    this.setState(
+      this._handleProps(nextProps.options.tickFormatSettings || {})
+    );
   }
 
-  _revert(axis) {
-    this._update(axis, this.state.initial[axis], null);
+  _handleProps(settings) {
+    return update(defaultTickFormatSettings, { $merge: settings });
   }
 
-  _handleChange(evt) {
-    const field = evt.target.name.split('.');
-    const value = ('number' === evt.target.type) ?
-      parseFloat(evt.target.value) : evt.target.value;
-    this.setState({ formatterSettings: update(this.state.formatterSettings,
-      { $apply: (settings) => {
-        settings[field[0]][field[1]] = value; // eslint-disable-line no-param-reassign
-        return settings;
-      } }
-    ) });
+  _localeOptions() {
+    return locales.map((locale, idx) => ({
+      children: `${locale.emoji} ${locale.name}`,
+      value: idx,
+    }));
   }
 
-  _applyFormatting(axis) {
-    this._update(
-      axis,
-      createFormatter(this.state.formatterSettings[axis]),
-      this.state.formatterSettings[axis]
-    );
+  /**
+   * Return full tickFormatSettings object after any element is changed
+   */
+  _handleChange(fieldProps, value) {
+    const field = fieldProps.name.split('.').pop();
+    return {
+      tickFormatSettings: update(this.state, { [field]: { $set: value } }),
+    };
   }
 
   render() {
-    const revertY = () => this._revert('y');
-    const applyFormattingY = () => this._applyFormatting('y');
     return (
       <div>
-        <ListBlock list={this.formatGuidelines} />
-        <div className={styles.fieldGroup}>
-          <Input
-            label="Prepend"
-            name="y.prepend"
-            rounded
-            value={this.state.formatterSettings.y.prepend}
-            onChange={this._handleChange}
+        <div>
+          <DispatchField
```

Comment:
I'm seeing lots of repetition and/or static configs here.
I would suggest DRYing this up, which the best way would probably be a config file which setups field properties.

See https://github.com/alleyinteractive/kff/blob/interactive/subsidy-penalty-calculator/build/apps/subsidy-penalty-calculator/app/config/penalty/localization_en.js#L5-L38

and implimentation: https://github.com/alleyinteractive/kff/blob/interactive/subsidy-penalty-calculator/build/apps/subsidy-penalty-calculator/app/components/formField/index.js